src/metabase/driver/sql/query_processor.clj

```diff
@@ -267,13 +271,24 @@
   [driver [_ _ expression-definition]]
   (->honeysql driver expression-definition))
 
+(defn unix-timestamp->honeysql*
```

Comment:
Star usually denotes an implementation function, while here it's a wrapper. Perhaps it would be cleaner to just package unit inference into own function and pass that to the OG `unix-timestamp->honeysql`. eg `special-type->unix-timestamp-unit`.